### PR TITLE
Feat/1750 failed deployment

### LINKED ISSUE #1750: [Incident Response Drill] As a developer, I need to know how to troubleshoot a failed deploy
During Sprint 44, we experienced a number of failed deployments, each of which appear to be for different reasons. As we're hardening our gitflow from development through production, it is important to document the circumstances around the failures,  root cause(s), how these were resolved, and lessons learned (i.e. post-mortem). This will help guide TDP developers in troubleshooting any future, similar incidents, if the need arises. 

**ACs**
- [x] post-mortem documented
- [x] any deployment-related READMEs are updated, if necessary

**Supporting Documentation:**
 - [TTS example of incident response for failed deployment](https://engineering.18f.gov/security/incident-response-drills/#scenario-a-deploy-goes-wrong)

### PR DESCRIPTION
**Pull request by andrew-jameson**
## Summary of Changes
_Provide a brief summary of changes_
Pull request closes #1750 

## How to Test
N/A

## Deliverables
_More details on how deliverables herein are assessed included [here](../docs/How-We-Work/our-priorities-values-expectations.md#Deliverables)._

### [Deliverable 1: Accepted Features](../docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-1-Accepted-Features)

Checklist of ACs:
+ [ ] [**_insert ACs here_**]
+ [ ] **`lfrohlich`** and/or **`adpennington`**  confirmed that ACs are met.

### [Deliverable 2: Tested Code](../docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-2-Tested-Code)

+ Are all areas of code introduced in this PR meaningfully tested?
  + [ ] If this PR introduces backend code changes, are they meaningfully tested?
  + [ ] If this PR introduces frontend code changes, are they meaningfully tested?
+ Are code coverage minimums met?
  + [ ] Frontend coverage: [_insert coverage %_] (see `CodeCov Report` comment in PR)
  + [ ] Backend coverage: [_insert coverage %_] (see `CodeCov Report` comment in PR)

### [Deliverable 3: Properly Styled Code](../docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-3-Properly-Styled-Code)

+ [ ] Are backend code style checks passing on CircleCI?
+ [ ] Are frontend code style checks passing on CircleCI?
+ [ ] Are code maintainability principles being followed?

### [Deliverable 4: Accessible](../docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-4-Accessibility)

+ [ ] Does this PR complete the epic? 
+ [ ] Are links included to any other gov-approved PRs associated with epic?
+ [ ] Does PR include documentation for Raft's a11y review? 
+ [ ] Did automated and manual testing with `iamjolly` and `ttran-hub` using Accessibility Insights reveal any errors introduced in this PR?

### [Deliverable 5: Deployed](../docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-5-Deployed)

+ [ ] Was the code successfully deployed via automated CircleCI process to development on Cloud.gov?

### [Deliverable 6: Documented](../docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-6-Code-documentation)

+ [ ] Does this PR provide background for why coding decisions were made?
+ [ ] If this PR introduces backend code, is that code easy to understand and sufficiently documented, both inline and overall?
+ [ ] If this PR introduces frontend code, is that code easy to understand and sufficiently documented, both inline and overall?
+ [ ] If this PR introduces dependencies, are their licenses documented?
+ [ ] Can reviewer explain and take ownership of these elements presented in this code review?

### [Deliverable 7: Secure](../docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-7-Secure)

+ [ ] Does the OWASP Scan pass on CircleCI?
+ [ ] Do manual code review and manual testing detect any new security issues?
+ [ ] If new issues detected, is investigation and/or remediation plan documented? 

### [Deliverable 8: User Research](../docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-8-User-Research)

Research product(s) clearly articulate(s):
+ [ ] the purpose of the research
+ [ ] methods used to conduct the research 
+ [ ] who participated in the research
+ [ ] what was tested and how
+ [ ] impact of research on TDP
+ [ ] (_if applicable_) final design mockups produced for TDP development

### CODE REVIEW

File: docs/Technical-Documentation/Failed-Deployment-Troubleshooting.md [hunk 1]
```diff
@@ -0,0 +1,80 @@
+# Failed Deployment Flow Troubleshooting
+
+Date: 2021-07-15
+
+## Status
+
+In Review
```
Comment (ADPennington) [inline docs/Technical-Documentation/Failed-Deployment-Troubleshooting.md:7]:
```suggestion
Accepted
```

File: docs/Technical-Documentation/Failed-Deployment-Troubleshooting.md [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+
+### Check that all environment variables exist in the backend app.
+1. Run the following command to see env variables for the target application: `cf env tdp-backend-<target>|less`
+2. Compare against other backend apps within cloud.gov to ensure the values are similar and that we are not missing any. You can also reference `deploy-backend.sh` for those that are set automatically.
+
+
+### Check that all services, routes are bound.
+You can reference the TDP diagram to find the relevant services [here](images/tdp-environments.png).
+
+Routes for dev environments should only be `tdp-frontend.apps.cloud.gov`, `tdp-backend-prod.apps.cloud.gov`, and the clamav-rest route.
```
Comment (ADPennington) [inline docs/Technical-Documentation/Failed-Deployment-Troubleshooting.md:47]:
```suggestion
Routes for dev environments should only be `tdp-frontend-<dev env>.apps.cloud.gov`, `tdp-backend-<dev env>.apps.cloud.gov`, and the clamav-rest route.
```

**Review by ADPennington — CHANGES_REQUESTED** [review]
this looks great @andrew-jameson! 🚀  couple of minor fixes prior to merge

**Review by n0remac — APPROVED** [review]

**Review by ADPennington — APPROVED** [review]
🚀

**Pull request merged**